app/controllers/questions_controller.rb

```diff
@@ -11,16 +11,29 @@ def create
         redirect_to questions_path, notice: "質問を受け付けました。"
       else
         @questions = Question.order(id: :desc)
-        flash.now[:alert] = "記入漏れがあります。"
         render :index
       end
     end
 
     def show
-        @question = Question.find(params[:id])
-        @solution = Solution.new
+      @question = Question.find(params[:id])
+      @solution = Solution.new
+      if @question.views.find_by(user_id: current_user.id).blank?
+        view = @question.views.build(user_id: current_user.id)
+        view.save
```

Comment:
変数を使用していないので `create` でよいでしょう。

```suggestion
        @question.views.create(user_id: current_user.id)
```